### PR TITLE
adding in ttclid (tiktok call back id) to fields

### PR DESCRIPTION
**Pull request by coreyching**
_A summary of your pull request, including the what change you're making and why._

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/tiktok-conversions/reportWebEvent/index.ts [hunk 1]
```diff
@@ -84,6 +84,18 @@ const action: ActionDefinition<Settings, Payload> = {
           else: { '@path': '$.anonymousId' }
         }
       }
+    },
+    ttclid: {
+      label: "TikTok Click  ID",
+      description: "The value of ttclid used to match website visitor events with TikTok ads. The ttclid is valid for 7 days. See [Set up](https://ads.tiktok.com/marketing_api/docs?rid=4eezrhr6lg4&id=1681728034437121) ttclid for details.",
```
Comment (gpsamson) [inline packages/destination-actions/src/destinations/tiktok-conversions/reportWebEvent/index.ts:90]:
```suggestion
      description: "The value of the ttclid used to match website visitor events with TikTok ads. The ttclid is valid for 7 days. See [Set up ttclid](https://ads.tiktok.com/marketing_api/docs?rid=4eezrhr6lg4&id=1681728034437121) for details.",
```

File: packages/destination-actions/src/destinations/tiktok-conversions/reportWebEvent/index.ts [hunk 2]
```diff
@@ -110,7 +122,10 @@ const action: ActionDefinition<Settings, Payload> = {
             external_id: userData.hashedExternalId,
             phone_number: userData.hashedPhoneNumber,
             email: userData.hashedEmail
-          }
+          },
+          ad: {
+            callback: payload.ttclid
+          },
```
Comment (gpsamson) [inline packages/destination-actions/src/destinations/tiktok-conversions/reportWebEvent/index.ts:128]: Do you know if the request will fail if no `ttclid` value exists in the payload?

**Review by mshwery — APPROVED** [review]
lgmt pending Gabe's question

File: packages/destination-actions/src/destinations/tiktok-conversions/reportWebEvent/index.ts (hunk 2 above)
Comment (coreyching) [reply packages/destination-actions/src/destinations/tiktok-conversions/reportWebEvent/index.ts:128]: no it's not required! but if provided will help matching in tiktok

Comment (gpsamson) [reply packages/destination-actions/src/destinations/tiktok-conversions/reportWebEvent/index.ts:128]: Gotcha. In other words, the empty `ad` object will still accepted by TikTok when a ttclid is not set:

```
ad: {}
```

**Pull request merged**